csm_web/scheduler/views/section.py

```diff
@@ -779,3 +784,198 @@ def wotd(self, request, pk=None):
             return Response({}, status=status.HTTP_200_OK)
 
         raise PermissionDenied()
+
+
+    @action(detail=True, methods=["GET", "POST"], url_path='swap')
+    def swap_no_id(self, request, section_id=None, swap_id=None):
+        """
+        GET: Returns all relevant Swap objects for current user.
+            Request format:
+                { student_id: int }
+            Response format:
+                {
+                    receiver: [`List of Swap objects`],
+                    sender: [`List of Swap objects`]
+                }
+                Where each Swap object has the following format:
+                {
+                    swap_id: int,
+                    name: string,
+                    mentor_name: string,
+                    time: DateTime,
+                }
+
+        POST: Handles creating a new Swap instance when a Swap a requested or
+        processes an existing Swap request.
+            Request format:
+                { email: string }
+                `email` is the email of the student who will be receiving the swap request.
+            Response format:
+                {}
+                With status code of 201 if successful, 400 if not.
+        """
+        section = get_object_or_error(Section.objects, pk=section_id)
+        student_id = request.data.get("student_id", -1) if request.method != "GET" \
+            else request.query_params.get("student_id", -1)
+
+        # check if student_id is specified in request
+        if student_id == -1:
+            raise PermissionDenied("The `student_id` field in the request cannot be empty,"
+                                   "please specify student.")
+
+        # check if student exists
+        try:
+            student = get_object_or_error(Student.objects, id=student_id)
+        except Exception:
+            raise NotFound("No student found with id: " + str(student_id))
```

Comment:
Generally you should never be catching the generic `Exception` object, as this means that any unexpected errors would also be caught and we wouldn't know about it. Make all of the `except Exception:` blocks throughout more specific, for _only_ the kinds of exceptions you expect could happen as a result of user error.